### PR TITLE
Add more control over Marketplace availability

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Add new constant `GLPI_MARKETPLACE_ENABLE` to configure the availability of the Marketplace.
- 0 = Completely disabled
- 1 = CLI access only
- 2 = Web access only
- 3 = CLI and Web access allowed (Default)

As with other configuration options defined by constants, this can be overridden by package maintainers (`inc/downstream.php`) or GLPI administrators (`config/local_define.php`).
See: https://glpi-install.readthedocs.io/en/latest/install/index.html#files-and-directories-locations

### CODE REVIEW

File: src/Marketplace/Controller.php [hunk 1]
```diff
@@ -72,6 +77,16 @@ public static function getTypeName($nb = 0)
         return __('Marketplace');
     }
 
+    public static function isCLIAllowed()
```
Comment (cedric-anne) [inline src/Marketplace/Controller.php:80]:
```suggestion
    public static function isCLIAllowed(): bool
```

File: src/Marketplace/Controller.php [hunk 2]
```diff
@@ -72,6 +77,16 @@ public static function getTypeName($nb = 0)
...
+    {
+        return (GLPI_MARKETPLACE_ENABLE & self::MP_MODE_CLI_ONLY) === self::MP_MODE_CLI_ONLY;
+    }
+
+    public static function isWebAllowed()
```
Comment (cedric-anne) [inline src/Marketplace/Controller.php:85]:
```suggestion
    public static function isWebAllowed(): bool
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Please add summary PHPDoc for new constants and methods.

Except this, it is OK for me.

**Review by cedric-anne — APPROVED** [review]

File: front/marketplace.download.php [hunk 3]
```diff
@@ -39,6 +39,9 @@
 
 use Glpi\Marketplace\Controller as MarketplaceController;
 
+if (!MarketplaceController::isWebAllowed()) {
+   Html::displayRightError();
```
Comment (cedric-anne) [inline front/marketplace.download.php:43]:
```suggestion
    Html::displayRightError();
```

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**